### PR TITLE
adding ci plots

### PR DESCRIPTION
**Pull request by ncapak**
## Purpose 
 Add diagnostics and CI plots to package so that we can take advantage of ClimaAtmos diagnostics directly

closes https://github.com/CliMA/ClimaCoupler.jl/issues/654

## Content
 Added ci_plots pipeline to coupler_driver, the source was ClimaAtmos

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Review by LenkaNovak — APPROVED** [review]
Great job, @ncapak! I've added a couple of comments and refined the location of the output. I've also asked @juliasloan25 to take a look for a more impartial review. After that, I'm happy to merge this, and we can apply this more broadly in follow-up PRs. Thanks for your contribution! 🎉

File: experiments/AMIP/user_io/ci_plots.jl [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        output_path = output_path[1]
+        is_comparison = false
+    end
+
+    if is_comparison
+        MAX_NUM_COLS = length(output_path)
+        save_path = output_path[1]
+        output_name *= "_comparison"
+    else
+        save_path = plot_path
```
Comment (juliasloan25) [inline experiments/AMIP/user_io/ci_plots.jl:75]: Do we need both `output_path` and `plot_path` in this function? It seems like we could set `save_path = output_path` here, but I could be missing something :)

File: experiments/AMIP/user_io/ci_plots.jl [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    period = ""
+    if "10d" in available_periods
+        period = "10d"
+    elseif "1d" in available_periods
+        period = "1d"
+    elseif "12h" in available_periods
+        period = "12h"
+    end
+
+    available_periods = ClimaAnalysis.available_periods(simdirs[1]; short_name = short_names_2D[1], reduction)
```
Comment (juliasloan25) [inline experiments/AMIP/user_io/ci_plots.jl:187]: It seems like this line is duplicated from line 177

**Review by juliasloan25 — COMMENTED** [review]
This is great, Neeku, thank you! I left some comments - feel free to reach out if anything doesn't make sense or you want to discuss more :)

File: experiments/AMIP/user_io/ci_plots.jl [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            push!(summary_files, file_path)
+            vars_left_to_plot -= MAX_PLOTS_PER_PAGE
+            page += 1
+        end
+    end
+
+    # Save plots
+    output_file = joinpath(save_path, "$(output_name).pdf")
+
+    pdfunite() do unite
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl:140]: Thank you for the review, @juliasloan25 ! I should be able to answer some of these... hopefully. ;) Poppler_jll is what ClimaAtmos uses to create the combined pdf file, where all the diagnostics are saved and then uploaded as artifacts. We followed ClimaAtmos's choices for packages to minimize the overall dependencies.

File: experiments/AMIP/user_io/ci_plots.jl [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+const LARGE_NUM = typemax(Int)
+const LAST_SNAP = LARGE_NUM
+const FIRST_SNAP = -LARGE_NUM
+const BOTTOM_LVL = -LARGE_NUM
+const TOP_LVL = LARGE_NUM
+
+function Makie.get_tickvalues(yticks::Int, ymin, ymax)
+    return range(max(ymin, 0), ymax, yticks)
+end
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl:21]: I think this is an extension of the Makie function, so it's not used here but it is used internally. It's used in [ClimaAtmos](https://github.com/CliMA/ClimaAtmos.jl/blob/main/post_processing/ci_plots.jl#L97) as well.

File: experiments/AMIP/user_io/ci_plots.jl (hunk 1 above)
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl:75]: Ah, output path is where the files are saved, plot path is for plots. We can rename `output_path` to `file_path` to clarify this.

File: experiments/AMIP/user_io/ci_plots.jl (hunk 2 above)
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl:187]: this one is for [2d](url) but it was in the wrong place anyway - my bad! ;)

File: experiments/AMIP/user_io/ci_plots.jl [hunk 5]
```diff
@@ -0,0 +1,201 @@
+# this file follows the ClimaAtmos.Diagnostics and ci_plots interfaces
+import CairoMakie
+import CairoMakie.Makie
+import ClimaAnalysis
+import ClimaAnalysis: Visualize as viz
+import ClimaAnalysis: SimDir, slice_time, slice
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl]: how about CAN for ClimaAnalysis?

File: experiments/AMIP/user_io/ci_plots.jl (hunk 4 above)
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply experiments/AMIP/user_io/ci_plots.jl:16]: Yeah, just tried this without and it stumbles.

File: experiments/AMIP/user_io/ci_plots.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (juliasloan25) [reply experiments/AMIP/user_io/ci_plots.jl:133]: Okay! That makes sense, thank you :)

File: experiments/AMIP/user_io/ci_plots.jl (hunk 5 above)
[the remark this answers is not in the document]
Comment (juliasloan25) [reply experiments/AMIP/user_io/ci_plots.jl]: That sounds good! Thank you

**Review by juliasloan25 — APPROVED** [review]
Looks great!

**Pull request merged**